integrations/spark/spark-job-dataquality/spark-job-dataquality-app/src/main/java/com/thinkbiganalytics/spark/dataquality/checker/DataQualityChecker.java

```diff
@@ -20,96 +20,398 @@
-        return hiveContext;
+    /**
+     * Sets the active rules based on the available rule list. The flow attribute dq.active.rules is
+     * used to specify active rules. If this attribute does not exist, all available rules are
+     * used.<br>
+     * If any rule does not exist, an exception is thrown
+     * 
+     * @throws MissingRuleException
+     */
+    protected void setActiveRules() throws MissingRuleException {
+
+        String activeRuleStr = flowAttributes.getAttributeValue(DataQualityConstants.DQ_ACTIVE_RULES_ATTRIBUTE,
+                                                                DataQualityConstants.DEFAULT_DQ_ACTIVE_RULES_VALUE);
+
+        if (activeRuleStr.equals(DataQualityConstants.DEFAULT_DQ_ACTIVE_RULES_VALUE)) {
+            log.info("All available Data Quality Rules will be used");
+            ruleList = new ArrayList<>(availableRules.values());
+        } else {
+            String[] parsedRuleStr = activeRuleStr.split(",");
+
+            for (String str : parsedRuleStr) {
+                addDataQualityRule(getAvailableRule(str.trim()));
+            }
+        }
+    }
+
+    /**
+     * Main method that conducts the data quality check. This is done by getting row counts and
+     * iterate through each data quality check. The results summary is provided after all are
+     * executed. The results are written to log as well as hive
+     */
+    protected boolean doDataQualityChecks() {
+
+        boolean isSuccessful = false;
+        try {
+            String databaseName = flowAttributes.getAttributeValue(DataQualityConstants.CATEGORY_ATTRIBUTE);
+            String tableName = flowAttributes.getAttributeValue(DataQualityConstants.FEED_ATTRIBUTE);
+            String processing_dttm = flowAttributes.getAttributeValue(DataQualityConstants.PROCESSING_DTTM_ATTRIBUTE);
+
+            String feedTableName = tableName + DataQualityConstants.FEED_TABLE_SUFFIX;
+            String invalidTableName = tableName + DataQualityConstants.INVALID_TABLE_SUFFIX;
+            String validTableName = tableName + DataQualityConstants.VALID_TABLE_SUFFIX;
+            String whereClause = DataQualityConstants.PROCESSING_DTTM_COLUMN + " = '" + processing_dttm + "'";
+
+            setSourceRowCount();
+
+            long rowCount;
+            rowCount = getRowCount(databaseName, feedTableName, whereClause);
+            flowAttributes.addAttribute(DataQualityConstants.DQ_FEED_ROW_COUNT_ATTRIBUTE, String.valueOf(rowCount));
+
+            rowCount = getRowCount(databaseName, invalidTableName, whereClause);
+            flowAttributes.addAttribute(DataQualityConstants.DQ_INVALID_ROW_COUNT_ATTRIBUTE, String.valueOf(rowCount));
+
+            rowCount = getRowCount(databaseName, validTableName, whereClause);
+            flowAttributes.addAttribute(DataQualityConstants.DQ_VALID_ROW_COUNT_ATTRIBUTE, String.valueOf(rowCount));
+
+            // Execute Data Quality rules
+            boolean rulePass;
```

Comment:
Looks like rulePass should be initialized to true. Otherwise, if no rules are set, the overall check will still fail due to Line 203. Please modify if so.